### DISCUSSION TITLE
Extend curl_easy_getinfo() to know whether we connected to the proxy or directly to a noproxy bypassed host

### DISCUSSION THREAD

**Discussion by genotrance** [curl/curl#12669] [Ideas]
When a libcurl is configured with proxy details along with noproxy rules, there's no way to know whether it connected directly to the target host (if the noproxy rule applied to that host) or whether it connected to the proxy.

Today, `curl_easy_getinfo()` can tell you the IP:port of the server connected using `CURLINFO_PRIMARY_IP` and `CURLINFO_PRIMARY_PORT`. However, it is tedious to figure out which host was connected to - we need to resolve all IPs of the target host and the proxy and figure out which has PRIMARY_IP.

The ideal solution would get a boolean - whether libcurl connected directly or to the proxy.

A compromise that aligns with PRIMARY_IP would be to provide the PRIMARY_HOSTNAME which was connected to - either the host or the proxy.

**Comment by bagder** [comment]
It's a good idea I think!

**Reply by bagder** [reply]
See #12719 and feel free to comment/test
